test/integrationtests/LorisIntegrationTest.class.inc

```diff
@@ -556,5 +559,184 @@ abstract class LorisIntegrationTest extends TestCase
             );
         }
     }
+    /**
+     * Get the content of a React Element
+     *
+     * @param  string $ui the location of this ui (css selector)
+     * @return string
+     */
+    function getReactElementContent($ui)
+    {
+        $attempts = 0;
+        $err = null;
+        do {
+            try {
+                return $this->webDriver->executescript(
+                    "return document.querySelector('$ui').textContent"
+                );
+            } catch (Exception $e) {
+         //       echo 'Caught exception: ',  $e->getMessage(), "\n";
+                        $err = $e;
+                        $attempts++;
+                        sleep(1);
```

Comment:
doesn't selenium have a "wait for element" that would be more effective than trying and sleeping repeatly?